bin/app.rb

```diff
@@ -21,15 +22,52 @@ def execute_operation(options)
       s3_bucket.delete_bucket
     when 'list'
       store = Store.new
-      store.list_buckets
+      buckets = store.list_buckets
+      buckets.each { |name| puts name.colorize(:yellow) }
+    when 'copy'
+      raise S3StoreArgumentError, "S3StoreArgumentError:: insufficient arguments" if options.count != 2
+
+      copy_params = extract_params(options)
+      bucket_name = copy_params[:bucket]
+      s3_obj = S3Object.new(bucket_name)
+      s3_obj.perform_operation(copy_params)
     else
       raise S3StoreNoServiceError, "Invalid operation name. Try #{OPERATIONS.join(', ')}.".colorize(:red) if (operation.nil? || !OPERATIONS.include?(operation.downcase))
     end  
   end
+
+  def extract_params(options)
+    source, destination, bucket = nil, nil, nil
+
+    source_file = File.basename(options[0])
+
+    if options[0].match?(BUCKET_URI_REGEX) && options[1].match?(BUCKET_URI_REGEX) # copy bucket-to-bucket
+      source = extract_s3_bucket(options[0])
+      destination = extract_s3_bucket(options[1])
+      bucket = source
```

Comment:
Are you also handling the following scenario?
Bucket to bucket copy, for a different bucket compared to the source for the destination.

Reply:
yes

Reply:
Making some changes to the way we extract bucket and object information. Changing the regex to a more grouped one to be able to extract information from the s3 uri's clearly.